sdk/include/psdk/gdiplusmatrix.h

```diff
@@ -23,82 +23,92 @@ class Matrix : public GdiplusBase
 {
     friend class Pen;
     friend class Region;
+    friend class GraphicsPath;
 
   public:
     Matrix(const RectF &rect, const PointF *dstplg)
     {
-        status = DllExports::GdipCreateMatrix3(&rect, dstplg, &matrix);
+        lastStatus = DllExports::GdipCreateMatrix3(&rect, dstplg, &nativeMatrix);
     }
 
     Matrix(const Rect &rect, const Point *dstplg)
     {
-        status = DllExports::GdipCreateMatrix3I(&rect, dstplg, &matrix);
+        lastStatus = DllExports::GdipCreateMatrix3I(&rect, dstplg, &nativeMatrix);
     }
 
-    Matrix(VOID)
+    Matrix()
     {
-        status = DllExports::GdipCreateMatrix(&matrix);
+        lastStatus = DllExports::GdipCreateMatrix(&nativeMatrix);
     }
 
     Matrix(REAL m11, REAL m12, REAL m21, REAL m22, REAL dx, REAL dy)
     {
-        status = DllExports::GdipCreateMatrix2(m11, m12, m21, m22, dx, dy, &matrix);
+        lastStatus = DllExports::GdipCreateMatrix2(m11, m12, m21, m22, dx, dy, &nativeMatrix);
     }
 
-    Matrix *Clone(VOID)
+    Matrix *
+    Clone()
     {
-        Matrix *cloneMatrix = new Matrix(); // FIXME: Matrix::matrix already initialized --> potential memory leak
-        cloneMatrix->status = DllExports::GdipCloneMatrix(matrix, cloneMatrix ? &cloneMatrix->matrix : NULL);
-        return cloneMatrix;
+        GpMatrix *cloneMatrix = NULL;
+        SetStatus(DllExports::GdipCloneMatrix(nativeMatrix, &cloneMatrix));
+
+        if (lastStatus != Ok)
+            return NULL;
+
+        return new Matrix(cloneMatrix);
     }
 
-    ~Matrix(VOID)
+    ~Matrix()
     {
-        DllExports::GdipDeleteMatrix(matrix);
+        DllExports::GdipDeleteMatrix(nativeMatrix);
     }
 
     BOOL
     Equals(const Matrix *matrix)
     {
         BOOL result;
-        SetStatus(DllExports::GdipIsMatrixEqual(this->matrix, matrix ? matrix->matrix : NULL, &result));
+        SetStatus(DllExports::GdipIsMatrixEqual(nativeMatrix, matrix ? matrix->nativeMatrix : NULL, &result));
         return result;
     }
 
     Status
     GetElements(REAL *m) const
     {
-        return SetStatus(DllExports::GdipGetMatrixElements(matrix, m));
+        return SetStatus(DllExports::GdipGetMatrixElements(nativeMatrix, m));
     }
 
-    Status GetLastStatus(VOID)
+    Status
+    GetLastStatus()
```

Comment:
can this get a "const" (because it doesn't modify the source object)?

Reply:
OK.